src/common/src/util/epoch.rs

```diff
@@ -97,6 +97,31 @@ impl fmt::Display for Epoch {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct EpochPair {
+    pub curr: u64,
+    pub prev: u64,
+}
+
+impl EpochPair {
+    pub fn new(curr: u64, prev: u64) -> Self {
+        assert!(curr > prev);
```

Comment:
Looks like no one is using this function that covers the assertion, as the raw epoch is made public. 🤣

Reply:
will take a look at it later :rofl: